toolkit/tools/pkg/imagecustomizerlib/customizepartitionsuuids.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/imagegen/diskutils"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safeloopback"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safemount"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+)
+
+func resetPartitionsUuids(buildImageFile string, buildDir string) error {
+	logger.Log.Infof("Resetting partition UUIDs")
+
+	loopback, err := safeloopback.NewLoopback(buildImageFile)
+	if err != nil {
+		return err
+	}
+	defer loopback.Close()
+
+	partitions, err := diskutils.GetDiskPartitions(loopback.DevicePath())
+	if err != nil {
+		return err
+	}
+
+	// Update the UUIDs.
+	newUuids := make([]string, len(partitions))
+	for i, partition := range partitions {
+		if partition.Type != "part" {
+			continue
+		}
+
+		newUuid, err := resetFileSystemUuid(partition)
+		if err != nil {
+			return fmt.Errorf("failed to reset partition's (%s) filesystem (%s) UUID:\n%w", partition.Path,
+				partition.FileSystemType, err)
+		}
+
+		newUuids[i] = newUuid
```

Comment:
should we assert that the uuid is set using `blkid` or so?

Reply:
Probably not. The entries under /dev are only updated asynchronously. So, the new value might not show up immediately.